### PR TITLE
Adding callback for after_empty_receive

### PR DESCRIPTION
**Pull request by bmalinconico**
I would like to add a callback for when an empty receive is encountered. I'm looking to use this callback to indicate that the worker node is preferred for termination during a scaling event.  Right now I'm detecting this using the before_request callback but it is convoluted.

I've added the code, and after I get a tentative 👍 for the feature I'll add documentation, and correct any changes you would like me to make.

I've named the callback after the SQS EmptyReceives metric measuring the same thing.

### CODE REVIEW

**Comment by mullermp** [comment]
Thanks for opening a pull request. Can you please share how you're doing this with `before_request`?

**Comment by bmalinconico** [comment]
@mullermp Thanks for the review.

Pesudocode for what I'm using right now would be:

```

poller.before_request do 
  empty_request_detected unless received_work

  received_work = false
end

poller.poll do |msgs|
  received_work = true 
end
```

It takes two empty receives to detect it using this method. First empty receive marks the received_work as false and a second empty receive will actually trigger it. Using the default 20s long polling configuration, it takes 40s to detect the worker is idle which is less than ideal.

**Comment by mullermp** [comment]
I think this makes sense to do. If you could add a CHANGELOG entry to aws-sdk-sqs and add documentation, that would be great.

**Review by mullermp — APPROVED** [review]

**Comment by bmalinconico** [comment]
Yep, I'll do that and add some documentation when I'm back in town next week

**Comment by bmalinconico** [comment]
@mullermp Documentation updated & CHANGELOG entry added

File: gems/aws-sdk-sqs/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Unreleased Changes
 ------------------
 
+* Feature - Update QueuePoller allow for the definition of an after_empty_receive callback which is triggered when a polling request returns no records.
```
Comment (mullermp) [inline gems/aws-sdk-sqs/CHANGELOG.md]: Please fix with:

```
* Feature - Update `QueuePoller` to allow for the definition of an `after_empty_receive` callback which is triggered when a polling request returns no records.
```

File: gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb [hunk 2]
```diff
@@ -180,6 +180,16 @@ module SQS
     #   end
     #   ```
     #
+    # * Configure a {#after_empty_receive} callback.
```
Comment (mullermp) [inline gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue_poller.rb]: a => an

**Review by mullermp — APPROVED** [review]
Thank you!

**Comment by mullermp** [comment]
Thank you for your contribution!

**Pull request merged**